src/NuGet.Core/NuGet.Protocol.Core.v3/Model/PackageSearchMetadataRegistration.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace NuGet.Protocol.Model
+{
+    public class PackageSearchMetadataRegistration : PackageSearchMetadata
```

Comment:
Add comments for this class. The name isn't very descriptive of adding a catalog uri, but I don't know of a better alternative.